src/gui/search/searchwidget.cpp

```diff
@@ -373,7 +373,7 @@ void SearchWidget::tabStatusChanged(QWidget *tab)
     {
         Q_ASSERT(m_activeSearchTab->status() != SearchJobWidget::Status::Ongoing);
 
-        if (app()->desktopIntegration()->isNotificationsEnabled() && (m_mainWindow->currentTabWidget() != this))
+        if (app()->desktopIntegration()->isNotificationsEnabled() && (app()->mainWindow()->currentTabWidget() != this))
         {
             if (m_activeSearchTab->status() == SearchJobWidget::Status::Error)
                 app()->desktopIntegration()->showNotification(tr("Search Engine"), tr("Search has failed"));
```

Comment:
I would remove this "wrong" dependency at all.
I don't insist on you do it now.

Reply:
I suppose you have plans for it. I'll leave it to you.

Reply:
> I suppose you have plans for it. I'll leave it to you.

#21816